build-with-ubp/action.yml

```diff
@@ -0,0 +1,65 @@
+name: 'Build with UBP'
+description: 'Creates a build with UBP'
+
+inputs:
+  files:
+    description: 'The files you want to build with space delimited'
+    required: true
+  override:
+    description: 'Your override json for the job'
+    required: true
+  platform:
+    description: 'The target platform'
+    required: true
+  unityVersion:
+    description: 'The version of unity for building'
+    required: true
+  buildWithUnityCore:
+    description: 'Toggle on to use Unity Core execute method'
+    default: false
+    required: false
+    type: boolean
+
+runs:
+  using: composite
+  steps:
+    - name: 'Run UBP'
+      env: 
+        UNITY_COMMAND: '/Applications/Unity/Hub/Editor/${{ inputs.unityVersion }}/Unity.app/Contents/MacOS/Unity -projectPath ${{ github.workspace }} -batchmode -logfile -'
+        EXECUTE_METHOD_UNITY_CORE: '-executeMethod JamCity.Core.Editor.UnifiedBuildPipeline.Commands.BuildPlayerFromFile'
+        EXECUTE_METHOD: '-executeMethod JamCity.UnityCore.UnifiedBuildPipelineSdk.Editor.Commands.BuildPlayerFromFile'
+        WITH_PARAMS: '-withParams ${{inputs.files}}'
+        OVERRIDE_FILE: 'override.json'
+        BUILD_TARGET: '-buildTarget'
```

Comment:
We could probably just inline some of these single-word values since they won't likely change anytime soon.

Reply:
I kept BUILD_TARGET since it makes the final command look nicer, but refactor a bunch of it